docs/admin/notifications/slack.md

```diff
@@ -0,0 +1,204 @@
+[Slack app](https://api.slack.com/apps), keeping your team updated on key events
+in your Coder environment.
+
+Administrators can configure Coder to send notifications via an incoming webhook
+endpoint. These notifications appear as conversation with a custom Slack app.
+
+## Requirements
+
+Before setting up Slack notifications, ensure that you have the following:
+
+- Administrator access to the Slack platform to create apps
+- Coder platform with notifications enabled
+
+## Create Slack Application
+
+To integrate Slack with Coder, follow these steps to create a Slack application:
+
+1. Go to the [Slack Apps](https://api.slack.com/apps) dashboard and create a new
+   Slack App.
+
+2. Under "Basic Information," you'll find a "Signing Secret." Keep it safe for
+   verification if needed.
+
+3. Under "OAuth & Permissions", add the following OAuth scopes:
+
+- `chat:write`: To send messages as the app.
+- `users:read`: To find the user details.
+- `users:read.email`: To find user emails.
+
+4. Install the app to your workspace and note down the **Bot User OAuth Token**
+   from the "OAuth & Permissions" section.
+
+## Build a Webserver to Receive Webhooks
+
+The Slack bot for Coder runs as a _Bolt application_, which is a framework
+designed for building Slack apps using the Slack API.
+[Bolt for JavaScript](https://github.com/slackapi/bolt-js) provides an
+easy-to-use API for responding to events, commands, and interactions from Slack.
+
+To build the server to receive webhooks and interact with Slack:
+
+1. Initialize your project by running:
+
+```bash
+npm init -y
+```
+
+2. Install the Bolt library:
+
+```bash
+npm install @slack/bolt
+```
+
+3. Create and edit the `app.js` file. Below is an example of the basic
+   structure:
+
+```js
+const { App, LogLevel, ExpressReceiver } = require("@slack/bolt");
+const bodyParser = require("body-parser");
+
+const port = process.env.PORT || 6000;
+
+// Create a Bolt Receiver
+const receiver = new ExpressReceiver({
+	signingSecret: process.env.SLACK_SIGNING_SECRET,
+});
+receiver.router.use(bodyParser.json());
+
+// Create the Bolt App, using the receiver
+const app = new App({
+	token: process.env.SLACK_BOT_TOKEN,
+	logLevel: LogLevel.DEBUG,
+	receiver,
+});
+
+receiver.router.post("/webhook", async (req, res) => {
+	try {
+		if (!req.body) {
+			return res.status(400).send("Error: request body is missing");
+		}
+
+		const { title, body } = req.body;
+		if (!title || !body) {
+			return res.status(400).send('Error: missing fields: "title", or "body"');
+		}
+
+		const payload = req.body.payload;
+		if (!payload) {
+			return res.status(400).send('Error: missing "payload" field');
+		}
+
+		const { user_email, actions } = payload;
+		if (!user_email || !actions) {
+			return res
+				.status(400)
+				.send('Error: missing fields: "user_email", "actions"');
+		}
+
+		// Get the user ID using Slack API
+		const userByEmail = await app.client.users.lookupByEmail({
```

Comment:
My JS knowledge is 10 years stale; does `await` raise an error, being caught by `catch`?

Reply:
Yes, it is in try-catch. Keep in mind that this is a minimal source for a Slack bot, on an official Coder product. Admins can always build their own bots looking our one as an example.